Add instructions for SSL certificates in quickstart (#826)
snowplow/documentation: docs/getting-started-on-community-edition/quick-start/index.md

1. Navigate to Amazon Certificate Manager (ACM) in the AWS Console
2. Request a public certificate from the ACM portal for the domain you want to host these endpoints under (e.g. for the Collector this might be `c.acme.com`) - make sure you are in the *same* region as your pipeline
  - `Fully qualified domain name` will be something like `c.acme.com`
  - `Validation method` is whatever works best for you - generally `DNS validation` is going to be easiest to action
  - `Key algorithm` should be left as `RSA 2048`
3. Once you have requested the certificate it should show up in the ACM certificate list as `Pending validation` - complete the DNS / Email validation steps and wait until the status changes to `Issued`
4. Copy the issued certificates `ARN` and copy it into your `terraform.tfvars` file under `ssl_information.certificate_arn`
5. Change `ssl_information.enabled` to `true`
6. Apply the `iglu_server` / `pipeline` module as you have done previously to attach the certificate to the Load Balancer
7. Add a `CNAME` DNS record for your requested domain pointing to the AWS Load balancer (e.g. `c.acme.com` -> `<lb-identifier>.<region>.elb.amazonaws.com`)

>   - If you are using `Route53` for DNS record management you can instead [setup an Alias route](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html) which can help circumvent certain `CNAME` cloaking tracking protections

Reviewer: ```suggestion
  - If you are using `Route53` for DNS record management, you can instead [setup an Alias route](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-to-elb-load-balancer.html) which can help circumvent certain `CNAME` cloaking tracking protections
```

> You should now be able to access your service over HTTPS - validate this by going to your newly setup endpoint in a browser and verifying that a valid response comes back and that the SSL certificate is valid.

Reviewer: ```suggestion
You should now be able to access your service over HTTPS. Verify this by going to your newly set up endpoint in a browser — you should get a valid response with a valid SSL certificate.
```

  </TabItem>
  <TabItem value="gcp" label="GCP">

1. Navigate to Google Certificate Manager in the GCP Console
2. Select the `Classic Certificates` tab and then press `Create SSL Certificate`
  - `Name` should be something unique for the certificate you are creating
  - `Description` describe the service you are creating the certificate for

>   - `Create mode` allows you to either supply your own certificate or create a `Google-managed` - we would generally recommend the later for ease of implementation and updates

Reviewer: ```suggestion
  - `Create mode` allows you to either supply your own certificate or create a `Google-managed` one. We generally recommend the latter for ease of implementation and updates
```

  - `Domains` add the domain you want to host the service under (e.g. for the Collector this might be `c.acme.com`)
3. Add an `A` DNS record for the requested domain pointing to the GCP Load balancer (e.g. `c.acme.com` -> `<lb-ipv4-address>`)
4. Copy the Certificate ID into your `terraform.tfvars` file under `ssl_information.certificate_id`
  - The ID takes the form `projects/{{project}}/global/sslCertificates/{{name}}`
5. Change `ssl_information.enabled` to `true`
6. Apply the `iglu_server` / `pipeline` module as you have done previously to attach the certificate to the Load Balancer

> Once the certificate is issued you should be able to access your service over HTTPS - validate this by going to your newly setup endpoint in a browser and verifying that a valid response comes back and that the SSL certificate is valid.

Reviewer: ```suggestion
Once the certificate is issued, you should now be able to access your service over HTTPS. Verify this by going to your newly set up endpoint in a browser — you should get a valid response with a valid SSL certificate.
```

:::info

> Its worth noting that GCP Managed Certificates can take up to 24 hours to be provisioned successfully so it might take a while before you can access your service over HTTPS.

Reviewer: ```suggestion
It’s worth noting that GCP Managed Certificates can take up to 24 hours to be provisioned successfully, so it might take a while before you can access your service over HTTPS.
```

:::

  </TabItem>
  <TabItem value="azure" label="Azure">

> Azure doesn't have a Certificate Management layer like AWS & GCP so to follow this guide you will need to have bought your own valid SSL certificate for use which covers the domain you want to bind onto.

Reviewer: ```suggestion
Unlike AWS or GCP, Azure doesn't have a Certificate Management layer. So to follow this guide, you will need to have bought your own valid SSL certificate covering the domain(s) you want to bind to.
```

> 1. [Convert your SSL Certificate](https://github.com/snowplow-devops/terraform-azurerm-lb?tab=readme-ov-file#adding-a-custom-certificate) into the correct format needed for the Azure Application Load Balancer
>   - Ultimately this requires you to convert your certificate into a `pkcs12` format

Reviewer: ```suggestion
1. [Convert your SSL Certificate](https://github.com/snowplow-devops/terraform-azurerm-lb?tab=readme-ov-file#adding-a-custom-certificate) into the correct format (`pkcs12`) needed for the Azure Application Load Balancer
```

2. Copy the `pkcs12` certificate password into your `terraform.tfvars` file under `ssl_information.password`
3. Copy the `pkcs12` certificate into your `terraform.tfvars` file under `ssl_information.data`

>  - Ensure the certificate is `base64` encoded (e.g. `cat cert.p12 | base64`)

Reviewer: ```suggestion
 - Ensure the certificate is `base64`-encoded (e.g. `cat cert.p12 | base64`)
```

4. Change `ssl_information.enabled` to `true`
5. Apply the `iglu_server` / `pipeline` module as you have done previously to attach the certificate to the Load Balancer
6. Add an `A` DNS record for your requested domain pointing to the Azure Load Balancer (e.g. `c.acme.com` -> `<lb-ipv4-address>`)

> You should now be able to access your service over HTTPS - validate this by going to your newly setup endpoint in a browser and verifying that a valid response comes back and that the SSL certificate is valid.

Reviewer: ```suggestion
You should now be able to access your service over HTTPS. Verify this by going to your newly set up endpoint in a browser — you should get a valid response with a valid SSL certificate.
```
